homeassistant/components/elmax/sensor.py

```diff
@@ -0,0 +1,71 @@
+"""Elmax sensor platform."""
+from __future__ import annotations
+
+from elmax_api.model.panel import PanelStatus
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ElmaxCoordinator
+from .common import ElmaxEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Elmax sensor platform."""
+    coordinator: ElmaxCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    known_devices = set()
+
+    def _discover_new_devices():
+        panel_status: PanelStatus = coordinator.data
+        # In case the panel is offline, its status will be None. In that case, simply do nothing
+        if panel_status is None:
+            return
+
+        # Otherwise, add all the entities we found
+        entities = []
+        for zone in panel_status.zones:
+            entity = ElmaxSensor(
+                panel=coordinator.panel_entry,
+                elmax_device=zone,
+                panel_version=panel_status.release,
+                coordinator=coordinator,
+            )
+            if entity.unique_id not in known_devices:
+                entities.append(entity)
```

Comment:
How unique is your unique id if you need to double check here if there are no duplicates?

Reply:
This has nothing to do with ID uniqueness: in fact this integration does provide unique IDs. 

The portion of code you referenced lies in the `_discover_new_devices()` method. This method is registered as a listener for the data `ElmaxCoordinator`. So, everytime the coordinator updates its data, this method is executed. This allows us to add newly device (as they are connected to the panel) automatically. However, this also means that at every coordinator update, the listener will try to add the same entities again. That is where we need to check what we already added and what we did not, so that we only add new ones and avoid adding the same devices again. 

I hope it's a bit clearer now.